examples/if_this_then_that/ifttt_gmail.py

```diff
@@ -0,0 +1,159 @@
+            You will use this address in your applet, below.
+
+        WARNING: Using ngrok exposes your local web server to the internet. See the ngrok
+        documentation for more information: https://ngrok.com/docs
+
+    2) Set up your applet on the "If This Then That" website.
+        a) Sign up and sign into https://ifttt.com
+        b) Create an applet: https://ifttt.com/create
+        c) Set up your trigger.
+            1. Click "this".
+            2. Select "Gmail" as your service. If prompted, click "Connect",
+                select your Gmail account, and click “Allow” to provide permissions
+                to IFTTT for your email account. Click "Done".
+            3. Under "Choose a Trigger", select “Any new email in inbox".
+        d) Set up your action.
+            1. Click “that".
+            2. Select “Maker" to set it as your action channel. Connect to the Maker channel if prompted.
+            3. Click “Make a web request" and fill out the fields as follows. Remember your publicly
+                accessible URL from above (e.g., http://55e57164.ngrok.io) and use it in the URL field,
+                followed by "/iftttGmail" as shown below:
+
+                 URL: http://55e57164.ngrok.io/iftttGmail
+                 Method: POST
+                 Content Type: application/json
+                 Body: {"FromAddress":"{{FromAddress}}"}
+
+            5. Click “Create Action" then “Finish".
+
+    3) Test your applet.
+        a) Run this script at the command line: ./ifttt_gmail.py
+        b) On ifttt.com, on your applet page, click “Check now”. See that IFTTT confirms that the applet
+            was checked.
+        c) Send an email to the Gmail account in your recipe
+        d) On your IFTTT applet webpage, again click “Check now”. This should cause IFTTT to detect that
+            the email was received and send a web request to the ifttt_gmail.py script.
+        e) In response to the ifttt web request, Cozmo should roll off the charger, raise and lower
+            his lift, announce the email, and then show a mailbox image on his face.
+'''
+
+import asyncio
+import re
+import sys
+
+
+try:
+    from aiohttp import web
+except ImportError:
+    sys.exit("Cannot import from aiohttp. Do `pip3 install --user aiohttp` to install")
+
+import cozmo
+
+from common import IFTTTRobot
+
+
+app = web.Application()
+
+
+async def serve_gmail(request):
+    '''Define an HTTP POST handler for receiving requests from If This Then That.
+
+    You may modify this method to change how Cozmo reacts to the email
+    being received.
+    '''
+
+    json_object = await request.json()
+
+    # Extract the name of the email sender.
+    from_email_address = json_object["FromAddress"]
+
+    # Use a regular expression to break apart pieces of the email address
+    match_object = re.search(r'([\w.]+)@([\w.]+)', from_email_address)
+    email_local_part = match_object.group(1)
```

Comment:
Minor/NitPick/Don'tFixNow. I assume this just gets the bit before the @ symbol - so e.g. name.name0123456789 - is it possible to get the associated name (like when you get the email there is often a corresponding name like `<"john doe" john.doe0123456789@....>`